debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/ReplicatorIT.java

```diff
@@ -254,6 +255,110 @@ public void shouldReplicateContent() throws InterruptedException {
         assertThat(records.size()).isEqualTo(1);
         Object[] allExpectedNames = { "Sally Hamm" };
         assertThat(foundNames).containsOnly(allExpectedNames);
+
+    }
+
+    @Test
+    public void shouldNotReplicateSnapshot() throws InterruptedException {
+
+        // ------------------------------------------------------------------------------
+        // SET SNAPSHOT MODE TO NEVER
+        // ------------------------------------------------------------------------------
+
+        Testing.Print.disable();
+        // Update the configuration to add a collection filter ...
+        useConfiguration(config.edit()
+                .with(MongoDbConnectorConfig.MAX_FAILED_CONNECTIONS, 1)
+                .with(MongoDbConnectorConfig.COLLECTION_WHITELIST, "dbA.contacts")
+                .with(MongoDbConnectorConfig.SNAPSHOT_MODE, MongoDbConnectorConfig.SnapshotMode.NEVER)
+                .build());
+
+        TestHelper.cleanDatabase(primary, "dbA");
+
+        // ------------------------------------------------------------------------------
+        // ADD A DOCUMENT
+        // ------------------------------------------------------------------------------
+        // Add a document to the 'contacts' database ...
+        primary.execute("shouldCreateContactsDatabase", mongo -> {
+            Testing.debug("Populating the 'dbA.contacts' collection");
+
+            // Create a database and a collection in that database ...
+            MongoDatabase db = mongo.getDatabase("dbA");
+            MongoCollection<Document> contacts = db.getCollection("contacts");
+            InsertOneOptions insertOptions = new InsertOneOptions().bypassDocumentValidation(true);
+            contacts.insertOne(Document.parse("{ \"name\":\"Jon Snow\"}"), insertOptions);
+            assertThat(db.getCollection("contacts").countDocuments()).isEqualTo(1);
+        });
+
+        // Start the replicator ...
+        List<SourceRecord> records = new LinkedList<>();
+        Replicator replicator = new Replicator(context, replicaSet, records::add, (x) -> {});
+        Thread thread = new Thread(replicator::run);
+        thread.start();
+
+        // Sleep for 2 seconds ...
+        Thread.sleep(2000);
+
+        // Stop the replicator ...
+        replicator.stop();
```

Comment:
This restart didn't seem needed, so I removed it while doing some clean-up. If there actually was strong reasoning for doing it, please let me know and we can discuss. Thanks!